### ISSUE TITLE
Can ContextualTransformers also implement Transform with empty context?

### ISSUE THREAD

**Issue by nickcarenza** [bazaarvoice/jolt#301]
I am trying to update the jolt version in Apache Nifi from 0.0.21 to 0.0.24 because I want to use the modify-overwrite-beta operation. As you might expect, integration code for the pre 0.0.22 release would use the Transform interface primarily which the new Modifier class does not implement. 

However, since there is no use for the context argument in this case, I wonder why Modifier doesn't just implement Transform with an empty map for the context. If it did, one would only have to update the dependency version to get access to the new operations.

**Comment by milosimpson** [comment]
So Nifi works at the Transform class level and not the Chainr level?   

It should work with Chainr, it is how BazaarVoice and the Demo site use jolt.  http://jolt-demo.appspot.com/#inception

https://github.com/bazaarvoice/jolt/blob/557a9efccf1847498c056c7c122d860ec00aec93/jolt-core/src/main/java/com/bazaarvoice/jolt/Chainr.java#L174

Once you have initialized your chain of transforms, can pass it a context or not.

That said to answer your question, it seemed important to keep Transform and ContextualTransform separate at the time.    With Java 1.8 and default method on an interface, I would be more ok with ContextualTransforms extending Transform, and just having that default method that forwards for you.

**Comment by nickcarenza** [comment]
It works at both. It has a dropdown that lets you select a single operation from a hardcoded list. If you select something other than Chainr, you only give it the spec.

I am using Chainr and passing an array of operations, one of which is modify-overwrite-beta. I didn't update the list of hardcoded values. Instead I just let it load the class based on ChainrEntry like it already was doing. After that though, it attempts to cast the class to the Transform interface since it is a [TransformFactory](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-utils/src/main/java/org/apache/nifi/processors/standard/util/TransformFactory.java) :). That's how they let you extend the functionality by providing your own class as long as it implements Transform.

**Comment by milosimpson** [comment]
Herm. 

So Chainr has an adapter to make all "regular" transforms look like ContextualTransforms.

At this point, I think all Transforms could just be Contextual, and just ignore the Context if it can't use it.

**Comment by nickcarenza** [comment]
Oh of course, you're right, that's a better idea.

This is the branch of Nifi I created to support modify-overwrite-beta https://github.com/nickcarenza/nifi/commit/9370120dea1e6810854cceac2d295c3d550da508. I added the wrapper class to call a ContextualTransform as if it were just a Transform. I was just trying to make as few changes in as few files as possible. Also I've never written Java before.
